File: Python/E-Labsheet/lab13/5.py
n = int(input(""))
char = input("")

if n < 0 or char not in ['e', 'E', 'o', 'O']:
    print("ERROR")
else:
    fib = [0] * (n + 1)
    fib[0] = 1
    if n > 0:
        fib[1] = 1
    for i in range(2, n + 1):
        fib[i] = fib[i - 1] + fib[i - 2]

    total = 0
    if char in ['e', 'E']:
        for i in range(0, n + 1, 2):
            total += fib[i]
    elif char in ['o', 'O']:
        for i in range(1, n + 1, 2):
            total += fib[i]

    print(total)

File: C/Lab/Lab4/5.c
#include <stdio.h>

int main() {
    int N;
    char input[10];
    fgets(input, sizeof(input), stdin);
    N = atoi(input);
    
    char top = 'a' + (N - 1);
    int total = 2 * N - 1;
    if( N <= 0 || N > 26){
        printf("-");
    }
    else{ 
        for (int i = 0; i < total; i++) {
        int r = (i < N) ? i : (2 * N - 2 - i);
        
        int s = 2 * (N - 1 - r);
        for (int d = 0; d < s; d++) {
            printf("-");
        }
        
        for (int k = 0; k <= r; k++) {
            printf("%c", (char)(top - k));
            if (k < r) {
                printf("-");
            }
        }
        
        for (int k = r - 1; k >= 0; k--) {
            printf("-%c", (char)(top - k));
        }
        
        for (int d = 0; d < s; d++) {
            printf("-");
        }
        
        printf("\n");
    }
    }
    
    return 0;
}
